src/test/java/teammates/common/datatransfer/attributes/NotificationAttributesTest.java

```diff
@@ -0,0 +1,269 @@
+
+        NotificationAttributes notificationAttributes =
+                NotificationAttributes.builder("notificationId")
+                .withStartTime(STARTTIME_ONE)
+                .withEndTime(ENDTIME_ONE)
+                .withType(NotificationType.DEPRECATION)
+                .withTargetUser(NotificationTargetUser.INSTRUCTOR)
+                .withTitle("valid notification title")
+                .withMessage("The first message")
+                .build();
+
+        notificationAttributes.update(updateOptions);
+
+        assertEquals(STARTTIME_TWO, notificationAttributes.getStartTime());
+        assertEquals(ENDTIME_TWO, notificationAttributes.getEndTime());
+        assertEquals(NotificationType.VERSION_NOTE, notificationAttributes.getType());
+        assertEquals(NotificationTargetUser.STUDENT, notificationAttributes.getTargetUser());
+        assertEquals("The edited title", notificationAttributes.getTitle());
+        assertEquals("The edited message", notificationAttributes.getMessage());
+    }
+
+    @Test
+    public void testUpdateOptionsBuilder_withNullInput_shouldFailWithAssertionError() {
+        assertThrows(AssertionError.class, () ->
+                NotificationAttributes.updateOptionsBuilder((String) null));
+        assertThrows(AssertionError.class, () ->
+                NotificationAttributes.updateOptionsBuilder("notificationId")
+                        .withStartTime(null));
+        assertThrows(AssertionError.class, () ->
+                NotificationAttributes.updateOptionsBuilder("notificationId")
+                        .withEndTime(null));
+        assertThrows(AssertionError.class, () ->
+                NotificationAttributes.updateOptionsBuilder("notificationId")
+                        .withType(null));
+        assertThrows(AssertionError.class, () ->
+                NotificationAttributes.updateOptionsBuilder("notificationId")
+                        .withTargetUser(null));
+        assertThrows(AssertionError.class, () ->
+                NotificationAttributes.updateOptionsBuilder("notificationId")
+                        .withTitle(null));
+        assertThrows(AssertionError.class, () ->
+                NotificationAttributes.updateOptionsBuilder("notificationId")
+                        .withMessage(null));
+    }
+
+    @Test
+    public void testEquals() {
+        NotificationAttributes notificationAttributes =
+                NotificationAttributes.builder("notificationId")
+                .withStartTime(STARTTIME_ONE)
+                .withEndTime(ENDTIME_ONE)
+                .withType(NotificationType.DEPRECATION)
+                .withTargetUser(NotificationTargetUser.INSTRUCTOR)
+                .withTitle("valid notification title")
+                .withMessage("valid message")
+                .build();
+
+        // When the two notifications are exact copies
+        NotificationAttributes notificationAttributesCopy = notificationAttributes.getCopy();
```

Comment:
This makes the test seem like you are testing `getCopy`. Probably should build another attribute object that is exactly the same and do the `equals`.